spec/test_dir/.gitignore

```diff
@@ -0,0 +1,4 @@
+# Ignore everything in this directory
```

Comment:
Why not use the gitignore in the top level directory and specify something along the lines of:
```
/spec/test_dir/*
```

Reply:
the dir needs to have something in it (even an invisible file like .gitignore) in order to get an otherwise empty directory in Git.

TBH I'm not super happy with using this dir at all. Thought at using the `/tmp` dir but XATTR could behave differently on /tmp with symlinks

> 
> The file permissions of symbolic links are not
> used in access checks.  These differences would allow users to
> consume filesystem resources in a way not controllable by disk quotas
>  for group or world writable special files and directories.
> 
> For this reason, user extended attributes are allowed only for
> regular files and directories, and access to user extended attributes
> is restricted to the owner and to users with appropriate capabilities
> for directories with the sticky bit set

Reply:
Alternatively a ".gitkeep" file could be used but it's fine. I thought you were trying to not commit anything in this directory but either works.

Reply:
.gitkeep  not a feature of Git, it just a trick to add an empty dir to a repo, the advantage in this case is that .gitognore also avoid adding any file to your commits